src/routing_table/mod.rs

```diff
@@ -694,4 +701,132 @@ mod tests {
+                    // We should get a few of these. Save one for tests, but otherwise ignore.
+                    unknown_distant_name = Some(new_name);
+                }
+                Err(_) => {
+                    assert!(false); // no other errors should be possible
+                }
+                Ok(true) => {
+                    assert!(table.our_group_prefix.matches(&new_name));
+                    assert!(unwrap!(table.groups.get(&table.our_group_prefix)).len() >=
+                            table.min_split_size() * 2);
+                    let our_prefix = *table.our_group_prefix();
+                    let _ = table.split(our_prefix);
+                    assert!(unwrap!(table.groups.get(&table.our_group_prefix)).len() >=
+                            table.min_split_size());
+                }
+                Ok(false) => {
+                    assert!(table.iter().any(|u| *u == new_name));
+                    if table.is_in_our_group(&new_name) {
+                        continue;   // add() already checked for necessary split
+                    }
+
+                    // Not a split event for our group, but might be for a different group.
+                    let group_prefix = table.find_group_prefix(&new_name)
+                        .expect("get group added to");
+                    let (group_len, new_group_size) = {
+                        let group = table.groups.get(&group_prefix).expect("get group from prefix");
+                        // Count size of group after an arbitrary split (note that there is only
+                        // one split possible; the arbitrariness is just which half we choose here).
+                        (group.len(),
+                         group.iter()
+                            .filter(|name| new_name.common_prefix(name) > group_prefix.bit_count())
+                            .count())
+                    };
+                    let min_size = table.min_split_size();
+                    if new_group_size >= min_size && group_len - new_group_size >= min_size {
+                        let _ = table.split(group_prefix);  // do the split
+                    }
+                }
+            }
+        }
+
+        let unknown_close_name;
+        loop {
+            let new_name = rng.next_u32();
+            if table.our_group_prefix.matches(&new_name) {
+                continue;
+            }
+            if let Some(prefix) = table.groups.keys().find(|p| p.matches(&new_name)) {
+                if !unwrap!(table.groups.get(&prefix)).contains(&new_name) {
+                    unknown_close_name = new_name;
+                    break;
+                }
+            }
+        }
+
+        let unknown_distant_name = unwrap!(unknown_distant_name);
+        // These numbers depend on distribution of names
+        let num_known_nodes = 114;
+        let num_groups = 9;
+        let len_our_group = 13;
```

Comment:
I wonder whether instead of a fixed seed, we shouldn't compute these numbers in the above iteration instead? (And possibly loop until `unknown_distant_name.is_some()`.
Or not assert them at all, but just bounds for them (e.g. we know that our own group must have at least `min_group_size` members, etc.).